Mathlib/Data/Multiset/Nodup.lean

```diff
@@ -137,6 +142,14 @@ theorem Nodup.map {f : α → β} {s : Multiset α} (hf : Injective f) : Nodup s
   Nodup.map_on fun _ _ _ _ h => hf h
 #align multiset.nodup.map Multiset.Nodup.map
 
+theorem nodup_map_iff_of_inj_on {f : α → β} (d : ∀ x ∈ s, ∀ y ∈ s, f x = f y → x = y) :
+    Nodup (map f s) ↔ Nodup s :=
+  ⟨Quot.induction_on s (fun _ ↦ List.Nodup.of_map _), fun h => h.map_on d⟩
+
+theorem nodup_map_iff_of_injective {f : α → β} (d : Function.Injective f) :
+    Nodup (map f s) ↔ Nodup s :=
+  ⟨Quot.induction_on s (fun _ ↦ List.Nodup.of_map _), fun h => h.map d⟩
```

Comment:
There's [Multiset.Nodup.of_map](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Multiset/Nodup.html#Multiset.Nodup.of_map) too.
And I wonder whether `∀ x ∈ s, ∀ y ∈ s, f x = f y → x = y` should be written as `Set.injOn f s` ...

Reply:
This was to match the signature of `Nodup.map_on` and `inj_on_of_nodup_map` etc. Maybe we should change all of them instead. But maybe in another PR?